third_party/nvfuser/csrc/scheduler/matmul.cpp

```diff
@@ -325,4 +334,310 @@ void scheduleMatmul(
   }
 }
 
+void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
+  const auto& inputs = fusion->inputs();
+  const auto& outputs = fusion->outputs();
+
+  TORCH_CHECK(
+      inputs.size() == 2,
+      "scheduleMatmul supports only fusions with two inputs");
+  TORCH_CHECK(
+      outputs.size() == 1,
+      "scheduleMatmul supports only fusions with single output");
+
+  TORCH_CHECK(
+      inputs[0]->isA<TensorView>(),
+      "fusion's first inpus is not an instance of TensorView class");
+  TORCH_CHECK(
+      inputs[1]->isA<TensorView>(),
+      "fusion's second inpus is not an instance of TensorView class");
+  TORCH_CHECK(
+      outputs[0]->isA<TensorView>(),
+      "fusion's output is not an instance of TensorView class");
+
+  TensorView* a = dynamic_cast<TensorView*>(inputs[0]);
```

Comment:
Check out https://github.com/csarofeen/pytorch/blob/devel/third_party/nvfuser/csrc/utils.h#L153-L235

You can just do `inputs[0]->as<TensorView>()` with our IR nodes

Reply:
Thanks for pointing this - I need to get more used to these tools so.
I fixed this in PR that I created in `NVIDIA/Fuser` project ([link](https://github.com/NVIDIA/Fuser/pull/23))